File: Juggler Sequence/Juggler_sequence.py
import math

num: int
while True:
    try:
        num = int(input("Enter a number: "))
        if num > -1:
            break
    except:
        print("Enter a positive integer: ")

# This function prints the juggler Sequence
def printJuggler(n):
    a = n

    # print the first term
    print(a)

    # calculate terms until last term is not 1
    while a != 1:
        b = 0

        # Check if previous term is even or odd
        if a % 2 == 0:

            # calculate next term
            b = (int)(math.floor(math.sqrt(a)))

        else:
            # for odd previous term calculate
            # next term
            b = (int)(math.floor(math.sqrt(a) * math.sqrt(a) * math.sqrt(a)))

        print(b)
        a = b


print(f"The juggler sequence for {num} is : ")
printJuggler(num)

File: Juggler Sequence/Juggler Sequence.cpp
// C++ implementation of Juggler Sequence
#include <iostream>
#include <cmath>
#include <math.h>
 
// This function prints the juggler Sequence
void printJuggler(long long n) {
    
    long long a = n;
 
    // print the first term
    std::cout << a << " ";
 
    // calculate terms until
    // last term is not 1
    while (a != 1)
    {
        long long b = 0;
 
        // Check if previous term
        // is even or odd
        if (!(a % 2)){
 
            // calculate next term
            b = floor(sqrt(a));
            
        }
 
        else{ 
 
            // for odd previous term
            // calculate next term
            b = floor(sqrt(pow(a, 3.0)));

        }
        std::cout << b << " ";
        a = b;
    }
}
 
// Driver Code
int main() {
    int n;
    
    std::cin >> n; // enter the starting number for the Sequence

    printJuggler(n);

}
